CMakeLists.txt

```diff
@@ -0,0 +1,713 @@
+            ${Silo_SOURCE_DIR}/src/hzip/hznio.cpp
+            ${Silo_SOURCE_DIR}/src/hzip/hznpredictor.cpp
+            ${Silo_SOURCE_DIR}/src/hzip/hznread.cpp
+            ${Silo_SOURCE_DIR}/src/hzip/hznwrite.cpp
+            ${Silo_SOURCE_DIR}/src/hzip/hzutil.cpp)
+
+        list(APPEND silo_library_include_dirs 
+            ${Silo_SOURCE_DIR}/src/hzip)
+    endif()
+
+    if(zlib_FOUND)
+        list(APPEND silo_library_include_dirs ${ZLIB_INCLUDE_DIR})
+    else()
+        list(APPEND SILO_COMPILE_DEFINES WITHOUT_ZLIB)
+    endif()
+endif()
+
+
+add_library(silo ${silo_library_sources})
+target_compile_definitions(silo PRIVATE ${SILO_COMPILE_DEFINES})
+add_dependencies(silo pdb_detect)
+target_include_directories(silo PRIVATE ${silo_library_include_dirs})
+
+# add list of public headers to be installed
+set(silo_headers
+    ${silo_build_include_dir}/lite_pdb.h
+    ${silo_build_include_dir}/lite_score.h
+    ${silo_build_include_dir}/pdform.h
+    ${silo_build_include_dir}/silo.h
+    ${Silo_SOURCE_DIR}/src/silo/silo.inc
+    ${Silo_SOURCE_DIR}/src/silo/silo_f9x.inc
+    ${Silo_SOURCE_DIR}/src/silo/silo_exports.h)
+
+if(WIN32)
+    list(APPEND silo_headers ${Silo_SOURCE_DIR}/src/silo/silo_win32_compatibility.h)
+    # Also install silodiff.bat into bin
+    install(FILES ${Silo_SOURCE_DIR}/resources/silodiff.bat DESTINATION bin
+            PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE
+                        GROUP_READ GROUP_WRITE GROUP_EXECUTE
+                        WORLD_READ             WORLD_EXECUTE)
+
+    if(SILO_ENABLE_SILEX OR SILO_ENABLE_BROWSER)
+        # copy silo dll to build/bin dir
+        add_custom_command(TARGET copy_deps POST_BUILD
+             COMMAND ${CMAKE_COMMAND} -E copy_if_different
+             $<TARGET_FILE:silo> ${Silo_BINARY_DIR}/bin/$<CONFIG>)
+    endif()
+endif()
+
+install(FILES ${silo_headers} DESTINATION include
+        PERMISSIONS OWNER_READ OWNER_WRITE
+                    GROUP_READ GROUP_WRITE
+                    WORLD_READ)
+
+
+if(SILO_ENABLE_HDF5 AND (TARGET hdf5 OR HDF5_FOUND))
+    # since we change the ouputname from silo to siloh5 when building
+    # with HDF5, will use $<TARGET_LINKER_FILE:silo> in place of plain 'silo'
+    # when telling broswer and silex to link with silo
+    set_target_properties(silo PROPERTIES OUTPUT_NAME siloh5)
+    if(TARGET hdf5)
+        target_link_libraries(silo hdf5)
+    else()
+        target_include_directories(silo PRIVATE ${HDF5_INCLUDE_DIRS})
+        target_link_libraries(silo ${HDF5_LIBRARIES})
+    endif()
+    if( zlib_FOUND)
+        target_link_libraries(silo zlib)
+    endif()
+endif()
+
+# setting up for installing Silo CMake config files
+set(silo_targets_name "Silo-targets")
+install(TARGETS silo EXPORT ${silo_targets_name}
+        PERMISSIONS OWNER_READ OWNER_WRITE
+                    GROUP_READ GROUP_WRITE
+                    WORLD_READ)
+
+##
+# silex
+##
+
+if(SILO_ENABLE_SILEX)
```

Comment:
Ok, so here we are not listing all the source files for Silex and instead just adding a dir. Why is this different from the Silo lib source files, above? Is the difference that there is a `CMakeLists.txt` file in the silex subdir that takes care of all that?

Reply:
Yes, I thought it better to put browser, silex, and pythonmodule CMakeLists.txt in their respective directories, since they create separate executables and library.  

Now that I think about it, the bulk of logic for creating the silo library could probably live in src/CMakeLists.txt

Reply:
Will keep silo lib setup as currently implemented in root CMakeLists.txt